db/db_filesnapshot.cc

```diff
@@ -126,8 +127,23 @@ Status DBImpl::GetSortedWalFiles(VectorLogPtr& files) {
            (pending_purge_obsolete_files_ > 0 || bg_purge_scheduled_ > 0)) {
       bg_cv_.Wait();
     }
+
+    // If the called didn't disable deletion, we should do it to avoid the case
```

Comment:
The problem is that another thread might have disabled file deletion and re-enable while this thread is expecting them to be disabled. For safety, I don't think it's a big deal to always disable (increment) and re-enable (decrement) considering how expensive GetSortedWalFiles is already (e.g. iterates over all SST files in the DB dir in most cases).

Reply:
I was hoping to minimize behavior change for normal path to minimize the risk. Now changed to the way you suggested.